### PR TITLE
Singularity beacon now attracts the Tesla Ball

### PR DESCRIPTION
**Pull request by Robustin**
Now kith.

### CODE REVIEW

File: code/game/machinery/syndicatebeacon.dm [hunk 1]
```diff
@@ -125,6 +125,9 @@
 	for(var/obj/singularity/singulo in world)
 		if(singulo.z == z)
 			singulo.target = src
+	for(var/obj/singularity/energy_ball/tesla in world)
```
Comment (RemieRichards) [inline code/game/machinery/syndicatebeacon.dm]: >in world

no

**Comment by KorPhaeron** [comment]
This doesn't even work

File: code/modules/power/tesla/energy_ball.dm [hunk 2]
```diff
@@ -33,6 +33,7 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
 	var/produced_power
 	var/energy_to_raise = 32
 	var/energy_to_lower = -20
+	var/home = null // Syndie Beacon target
```
Comment (ChangelingRain) [inline code/modules/power/tesla/energy_ball.dm]: Couldn't you use the inherited target var from the singularity?

File: code/game/machinery/syndicatebeacon.dm (hunk 1 above)
Comment (DaedalusGame) [inline code/game/machinery/syndicatebeacon.dm]: >complain about tesla in world
>singulo in world right above that
so when are you PRing a fix?

Comment (AnturK) [inline code/game/machinery/syndicatebeacon.dm]: You forgot to reset it on deactivation, also these could probably be reduced to in poi_list

Comment (ChangelingRain) [inline code/game/machinery/syndicatebeacon.dm]: This code isn't needed if robustin changes it to use target instead of a new var.

Comment (KorPhaeron) [inline code/game/machinery/syndicatebeacon.dm]: @DaedalusGame 

Shit oldcode isn't an excuse for adding shit newcode

File: code/modules/power/tesla/energy_ball.dm (hunk 2 above)
Comment (Robustin) [inline code/modules/power/tesla/energy_ball.dm]: Energy Ball has its own target obj, so I wanted to use home for clarity. I have no idea what gets inherited.

File: code/game/machinery/syndicatebeacon.dm (hunk 1 above)
Comment (Robustin) [inline code/game/machinery/syndicatebeacon.dm]: b...but it works for singularities?

File: code/modules/power/tesla/energy_ball.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/power/tesla/energy_ball.dm]: >I have no idea what gets inherited.

Everything

**Comment by Robustin** [comment]
Clarified in chat that FOR creates lists and shit and inheritance works based on the object path not the path of what defines the object.

My power swells.

File: code/modules/power/tesla/energy_ball.dm [hunk 3]
```diff
@@ -79,9 +79,12 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
 	var/first_move = dir
 	for(var/i in 0 to move_amount)
 		var/move_dir = pick(alldirs + first_move) //give the first move direction a bit of favoring.
+		if(target && prob(60))
+			move_dir = get_dir(src,target)
 		var/turf/T = get_step(src, move_dir)
 		if(can_move(T))
 			loc = T
+
```
Comment (xxalpha) [inline code/modules/power/tesla/energy_ball.dm]: reeeee

File: code/modules/power/tesla/energy_ball.dm [hunk 4]
```diff
@@ -33,7 +33,7 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
 	var/produced_power
 	var/energy_to_raise = 32
 	var/energy_to_lower = -20
-
+
```
Comment (PKPenguin321) [inline code/modules/power/tesla/energy_ball.dm]: >adding a random tab to a blank line for no reason

triggered

File: code/modules/power/tesla/energy_ball.dm [hunk 5]
```diff
@@ -79,9 +79,12 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
 	var/first_move = dir
 	for(var/i in 0 to move_amount)
 		var/move_dir = pick(alldirs + first_move) //give the first move direction a bit of favoring.
+		if(target && prob(60))
```
Comment (MrStonedOne) [inline code/modules/power/tesla/energy_ball.dm]: You need to make sure target is on the same zlevel as the tesla.

just do `if(target && target.z = z && prob(60))`

**Comment by PKPenguin321** [comment]
also make sure to add a changelog

**Comment by Iamgoofball** [comment]
:+1: thanks fam I keep forgetting to do this

**Comment by CPTANT** [comment]
Does it destroy the beacon on reaching it?

**Comment by PKPenguin321** [comment]
it's probably safe to assume that by the time it has reached the beacon, it has destroyed and power generating objects that wire into it

**Comment by SweaterKittens** [comment]
Especially given the Tesla's tendency to pick a spot and hang out forever, this is pretty cool and useful.

**Comment by Robustin** [comment]
It shouldn't be destroyed, which makes it a fun alternative to sing releasing since you can relocate the beacon relatively safely (assuming you have a hardsuit+gloves).

I added this primarily for consistency (will add something to the description soon to clarify it works on both engines) and to remove some of the blatant advantages Tesla engine has.

**Comment by ChangelingRain** [comment]
It'll get shocked and probably destroyed because it *is* a machine, but it'd be easy enough to override that.

**Comment by MrStonedOne** [comment]
its not a machine, in fact i don't even think it's a structure.

unless this was changed.

**Comment by ChangelingRain** [comment]
It's machinery/power, you're thinking of the powersink, which is an item.

**Pull request merged**